pennylane/estimator/ops/op_math/symbolic.py

```diff
@@ -0,0 +1,473 @@
+        algorithmic wires: 3
+        allocated wires: 0
+                 zero state: 0
+                 any state: 0
+     Total gates : 7
+      'SWAP': 1,
+      'ControlledPhaseShift': 3,
+      'Hadamard': 3
+    >>>
+    >>> print(qre.estimate(adj_qft, gate_set))
+    --- Resources: ---
+     Total wires: 3
+        algorithmic wires: 3
+        allocated wires: 0
+                 zero state: 0
+                 any state: 0
+     Total gates : 7
+      'Adjoint(ControlledPhaseShift)': 3,
+      'Adjoint(SWAP)': 1,
+      'Adjoint(Hadamard)': 3
+
+    """
+
+    resource_keys = {"base_cmpr_op"}
+
+    def __init__(self, base_op: ResourceOperator) -> None:
+        _dequeue(op_to_remove=base_op)
+        self.queue()
+        base_cmpr_op = base_op.resource_rep_from_op()
+
+        self.base_op = base_cmpr_op
+        self.wires = base_op.wires
+        self.num_wires = base_cmpr_op.num_wires
+
+    @property
+    def resource_params(self) -> dict:
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: A dictionary containing the resource parameters:
+             * base_cmpr_op (:class:`~.pennylane.estimator.ResourceOperator`): The operator
+               that we want the adjoint of.
+
+        """
+        return {"base_cmpr_op": self.base_op}
+
+    @classmethod
+    def resource_rep(cls, base_cmpr_op: CompressedResourceOp) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute a resource estimation.
+
+        Args:
+            base_cmpr_op (:class:`~.pennylane.estimator.ResourceOperator`): The operator
+                that we want the adjoint of.
+
+        Returns:
+            :class:`~.pennylane.estimator.resource_operator.CompressedResourceOp`: the operator in a compressed representation
+        """
+        num_wires = base_cmpr_op.num_wires
+        return CompressedResourceOp(cls, num_wires, {"base_cmpr_op": base_cmpr_op})
+
+    @classmethod
+    def resource_decomp(cls, base_cmpr_op: CompressedResourceOp, **kwargs):
+        r"""Returns a list representing the resources of the operator. Each object represents a
+        quantum gate and the number of times it occurs in the decomposition.
+
+        Args:
+            base_cmpr_op (:class:`~.pennylane.estimator.resource_operator.CompressedResourceOp`): A
+                compressed resource representation for the operator we want the adjoint of.
+
+        Resources:
+            This symbolic operation represents the adjoint of some base operation. The resources are
+            determined as follows. If the base operation implements the
+            :code:`.adjoint_resource_decomp()` method, then the resources are obtained from
+            this.
+
+            Otherwise, the adjoint resources are given as the adjoint of each operation in the
+            base operation's resources.
```

Comment:
```suggestion
            This symbolic operation represents the adjoint of some base operation. The resources are
            determined as follows. If the base operation implements the
            :code:`.adjoint_resource_decomp()` method, then the resources are obtained from
            this method. Otherwise, the adjoint resources are given as the adjoint of each operation in the
            base operation's resources.
```